### PR TITLE
Inversion of control

### PR DESCRIPTION
**Pull request by yarinvak**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,14 +5,21 @@
...
+    "merge-graphql-schemas": "^1.5.3",
+    "reflect-metadata": "^0.1.12"
   },
   "devDependencies": {},
   "scripts": {
     "test": "test",
     "linkPolaris": "npm link ../polaris",
-    "build": "npm install && npm run linkPolaris"
+    "build": "npm install && npm run linkPolaris && tsc",
+    "start": "node ./dist/app.js"
```
Comment (osher-sade) [inline package.json]: add clean script, and add a script that only compile the ts files. like this:
"clean": "rimraf dist"
"tsc": "tsc"

File: package.json [hunk 2]
```diff
@@ -5,14 +5,24 @@
   "main": "app.js",
   "dependencies": {
     "@enigmatis/polaris": "0.0.2",
+    "@types/glob": "^5.0.35",
+    "glob": "^7.1.2",
```
Comment (guy120494) [inline package.json:9]: We already have @types/glob. Delete this dependency

File: schema/schema.ts [hunk 3]
```diff
@@ -1,24 +1,28 @@
...
+
+// Create container
+let container = new Container();
+container.load(buildProviderModule());
+
+let creator: ISchemaCreator = container.get<ISchemaCreator>("ISchemaCreator");
+
+let schema = creator.generateSchema();
+
+export default schema
```
Comment (guy120494) [inline schema/schema.ts]: Dont use export default

File: schema/entities/book.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import "reflect-metadata";
+import {InjectableType} from '@enigmatis/polaris';
+import { provide} from "inversify-binding-decorators";
```
Comment (guy120494) [inline schema/entities/book.ts]: Delete the space between "{" and "provide"

File: package.json (hunk 2 above)
Comment (yarinvak) [reply package.json:9]: its required

**Review by osher-sade — APPROVED** [review]

**Pull request closed without merging**